gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -108,6 +123,156 @@ URLs - this aspect is taken care of in the preferences server component.
+        gpii.ontologyHandler.floydWarshallAlgorithmCore(data.dist, data.next, data.numVertices);
+        return data;
+    };
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction
+     * to initialize the mapping of vertices named by ontology to integers used in
+     * the algorithm.
+     *
+     * @data (Object) Data structure used in main function.
+     * @edges (Array) List of edges where each is an Object contains a directed source and
+     *      target parameter.
+     * @returns Does not a return a value, operates directly on the data structure.
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmInitVertices = function(data, edges) {
+        data.vertices = {};
+        fluid.each(edges, function(edge) {
+            data.vertices[edge.source] = true;
+            data.vertices[edge.target] = true;
+        });
+        // Assign each vertex an integer key for the lookup arrays
+        fluid.each(Object.keys(data.vertices), function(vert, i) {
+            data.vertices[vert] = i;
+        });
+        data.numVertices = Object.keys(data.vertices).length;
+    }
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction
+     * to initialize the core matrices used in the algorithm, next and dist. 
+     *
+     * @data (Object) Data structure used in main function.
+     * @edges (Array) List of edges where each is an Object contains a directed source and
+     *      target parameter.
+     * @returns Does not a return a value, operates directly on the data structure.
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmInitMatrices = function(data, edges) {
+        // Create |V|x|V| dist and next arrays and initialize to Infinity and null
+        data.dist = fluid.generate(data.numVertices, function() {
+            return fluid.generate(data.numVertices, Infinity, false);
+        }, true);
+        data.next = fluid.generate(data.numVertices, function() {
+            return fluid.generate(data.numVertices, null, false);
+        }, true);
+        fluid.each(edges, function(edge) {
+            var u = data.vertices[edge.source], 
+                v = data.vertices[edge.target];
+            data.dist[u][v] = 1; // All weights currently 1
+            data.next[u][v] = v;
+        });
+    };
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction
+     * to perform the core algorithm of Floyd-Warshall.
+     *
+     * @dist (2D array) See algorithm documentation
+     * @next (2D array) See algorithm documentation
+     * @numVertices (Integer) total number of vertices
+     * @returns Does not a return a value, operates directly on the dist and next matrices.
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmCore = function(dist, next, numVertices) {
```

Comment:
Perhaps for clarity all these can go in a sub-level namespace e.g.
gpii.ontologyHandler.floydWarshall.algorithmCore